packages/agent-core/src/factory.ts

```
@@ -42,13 +95,20 @@ export class AgentFactory {
-        mcpManager.initialize(serverConfigs).catch((error: Error) => {
-          logger.error("MCP manager initialization failed:", error);
-        });
+        try {
+          // Initialize MCP connections
+          await mcpManager.initialize(serverConfigs);
 
-        logger.debug(
-          `Agent created with ${serverConfigs.length} MCP servers: ${serverConfigs.map(s => s.name).join(", ")}`,
-        );
+          // Wait for MCP servers to be ready with proper timeout
+          await this.waitForMCPServersReady(mcpManager, serverConfigs, 10000);
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)